src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -81,6 +81,19 @@ void ColumnBetweenTableScanImpl::_scan_generic_segment(
       const auto typed_left_value = boost::get<ColumnDataType>(left_value);
       const auto typed_right_value = boost::get<ColumnDataType>(right_value);
 
+      // Filter out trivial cases here, as with_between_comparator assumes that lower and upper bound
+      // are set such that values could match the predicate.
+      if constexpr (std::is_integral_v<ColumnDataType>) {
+        using SignedColumnDataType = std::make_signed_t<ColumnDataType>;
```

Comment:
Why do we need that? Can't `ColumnDataType` be only `int32_t` or `int64_t` here (and thus, always be signed)?